doc/userdoc/nest_server.rst

```diff
@@ -0,0 +1,649 @@
+                  if (xhr.readyState === 4) {  // request done
+                      callback(xhr.responseText);
+                  }
+              });
+              // send to api route of NEST Server
+              xhr.open("GET", "http://localhost:5000/api/" + call);
+              xhr.send(null);
+          }
+
+Using the above code, we can already send API-requests to NEST Server:
+
+.. code-block:: JavaScript
+
+    getAPI('Models');  // a list of models
+
+Sending API calls with data requires a POST request, which can handle
+the data in JSON-format. To allow for this, we can define a function
+with a callback for POST requests:
+
+.. code-block:: JavaScript
+
+    function postAPI(call, data, callback=console.log) {
+        const xhr = new XMLHttpRequest();
+        xhr.addEventListener("readystatechange", () => {
+            if (xhr.readyState === 4) {  // request done
+                callback(xhr.responseText);
+            }
+        });
+        // send to api route of NEST Server
+        xhr.open("POST", "http://localhost:5000/api/" + call);
+        xhr.setRequestHeader('Access-Control-Allow-Headers', 'Content-Type');
+        xhr.setRequestHeader('Content-Type', 'application/json');
+        xhr.send(JSON.stringify(data));  // serialize data
+    }
+
+Using this function, sending an API-request to NEST Server becomes easy:
+
+.. code-block:: JavaScript
+
+    // default values of iaf_psc_alpha
+    postAPI('GetDefaults', {"model": "iaf_psc_alpha"});
+
+The third type of request we might want to make is sending a custom
+Python script to NEST Server. As outlined above, this is supported by
+the `exec` route. to make use of that, we define a function with
+callback for POST requests to execute a script:
+
+.. code-block:: JavaScript
+
+    function execScript(source, returnData="data", callback=console.log) {
+        const data = {"source": source, "return": returnData};
+        const xhr = new XMLHttpRequest();
+        xhr.addEventListener("readystatechange", () => {
+            if (xhr.readyState === 4) {  // request done
+                callback(xhr.responseText);
+            }
+        });
+        // send to exec route of NEST Server
+        xhr.open("POST", "http://localhost:5000/exec");
+        xhr.setRequestHeader('Access-Control-Allow-Headers', 'Content-Type');
+        xhr.setRequestHeader('Content-Type', 'application/json');
+        xhr.send(JSON.stringify(data));  // serialize data
+    }
+
+Now, we can send a custom Python script to NEST Server:
+
+.. code-block:: JavaScript
+
+    // default values of iaf_psc_alpha
+    execScript("data = nest.GetDefaults('iaf_psc_alpha')");
+
+.. note::
+
+    An full HTML client for NEST Server based on the ideas outlined
+    above is available in the `nest-jsclient repository
+    <https://github.com/steffengraber/nest-jsclient>`_ on the GitHub
+    account of Steffen Graber.
+
+Controlling NEST from Bash
```

Comment:
```suggestion
Control NEST from Bash
```
I'm also not a fan of the word control here. I also don't know what this implies.